packages/flutter/lib/src/rendering/box.dart

```diff
@@ -2362,14 +2361,26 @@ abstract class RenderBox extends RenderObject {
       _cachedBaselines?.clear();
       _cachedIntrinsicDimensions?.clear();
       _cachedDryLayoutSizes?.clear();
-      if (parent is RenderObject) {
-        markParentNeedsLayout();
-        return;
-      }
+      return true;
+    }
+    return false;
+  }
+
+  @override
+  void markNeedsLayout() {
+    if (_clearCachedData() && parent is RenderObject) {
+      markParentNeedsLayout();
+      return;
     }
     super.markNeedsLayout();
   }
 
+  @override
+  void layout(Constraints constraints, {bool parentUsesSize = false}) {
+    _clearCachedData();
```

Comment:
I'm writing a constraint layout for Flutter, where the size of the child elements is calculated by the parent element based on the constraints of the child element.  Child elements cannot determine their own size.  When the constraints of the child element change, the parent element will re-layout the child element. At this time, the child element should no longer notify the parent element of the layout.

Reply:
I'm writing a constraint layout for Flutter, where the size of the child elements is calculated by the parent element based on the constraints of the child element.  Child elements cannot determine their own size.  When the constraints of the child element change, the parent element will re-layout the child element. At this time, the child element should no longer notify the parent element of the layout.

Reply:
https://github.com/hackware1993/Flutter-ConstraintLayout

Reply:
`Clearing the intrinsics and drylayout cache here doesn't seem correct. These are usually called right before layout, so, if we clear them on layout, we will never benefit from the cache.`

Makes sense, but there is a doubt that if the RO self dirty, we will clear the cache data because we think the cached data may be needed an update. But if its ancestors were dirty, the cached data may also be inaccurate, right? This issue is a case of this scenario.

`Furthermore, this also makes us forget that a parent asked for our intrinsics. So next time we mark ourselves as needing layout, we will actually forget to tell our parent to mark itself as needing layout as well to ask for our new intrinsics again. If we don't have a test for this scenario, we should add one!`

Nice catch, I will add a test to cover this.

Reply:
Maybe we can clear the cache if the constraints changed, what is your thoughts of this? @goderbauer

Reply:
I think the best fix is to let constraints participate in the calculation of the _cachedBaselines key, what do you think?

Reply:
> I think the best fix is to let constraints participate in the calculation of the _cachedBaselines key, what do you think?

It will cause a lot of memory use when window resize.